umh-core/pkg/communicator/actions/deploy-dataflowcomponent.go

```diff
@@ -615,20 +615,36 @@ func (a *DeployDataflowComponentAction) waitForComponentToBeActive() error {
 						stateMessage := RemainingPrefixSec(remainingSeconds) + fmt.Sprintf("completed. is in state '%s' with correct configuration", instance.CurrentState)
 						SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionExecuting, stateMessage,
 							a.outboundChannel, models.DeployDataFlowComponent)
-						return nil
-					} else {
-						// currentStateReason contains more information on why the DFC is in its current state
-						currentStateReason := dfcSnapshot.ServiceInfo.StatusReason
-
-						stateMessage := RemainingPrefixSec(remainingSeconds) + currentStateReason
-						SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionExecuting, stateMessage, a.outboundChannel, models.DeployDataFlowComponent)
-						// send the benthos logs to the user
-						logs = dfcSnapshot.ServiceInfo.BenthosObservedState.ServiceInfo.BenthosStatus.BenthosLogs
-						// only send the logs that have not been sent yet
-						if len(logs) > len(lastLogs) {
-							lastLogs = SendLimitedLogs(logs, lastLogs, a.instanceUUID, a.userEmail, a.actionUUID, a.outboundChannel, models.DeployDataFlowComponent, remainingSeconds)
+						return "", nil
+					}
+
+					// currentStateReason contains more information on why the DFC is in its current state
+					currentStateReason := dfcSnapshot.ServiceInfo.StatusReason
+
+					stateMessage := RemainingPrefixSec(remainingSeconds) + currentStateReason
+					SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionExecuting, stateMessage, a.outboundChannel, models.DeployDataFlowComponent)
+					// send the benthos logs to the user
+					logs = dfcSnapshot.ServiceInfo.BenthosObservedState.ServiceInfo.BenthosStatus.BenthosLogs
+
+					// only send the logs that have not been sent yet
+					if len(logs) > len(lastLogs) {
+						lastLogs = SendLimitedLogs(logs, lastLogs, a.instanceUUID, a.userEmail, a.actionUUID, a.outboundChannel, models.DeployDataFlowComponent, remainingSeconds)
+					}
+					// check if the logs contain any of the error lines and if so, cancel the action with rolling back
+					if CheckBenthosLogLinesForConfigErrors(logs) {
+						SendActionReply(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionExecuting, "Failed to parse config. Removing the component.", a.outboundChannel, models.DeployDataFlowComponent)
+						ctx, cancel := context.WithTimeout(context.Background(), constants.ActionTimeout)
+						defer cancel()
+						err := a.configManager.AtomicDeleteDataflowcomponent(ctx, dataflowcomponentserviceconfig.GenerateUUIDFromName(a.name))
+						if err != nil {
+							a.actionLogger.Errorf("failed to remove dataflowcomponent %s: %v", a.name, err)
+							SendActionReplyV2(a.instanceUUID, a.userEmail, a.actionUUID, models.ActionFinishedWithFailure, "DFC not removed. Please check your configuration and consider removing the component manually.", "ERR_CONFIG_ERROR", nil, a.outboundChannel, models.DeployDataFlowComponent, nil)
```

Comment:
what does ERR_CONFIG_ERROR mean? what does it do? --> needs to be documented somewhere. either in SendActionReplyV2 or somewhere else